execution/src/posix/execution.cc

```diff
@@ -139,8 +142,33 @@ namespace leatherman { namespace execution {
         return fs.st_mode & S_IXOTH;
     }
 
-    string which(string const& file, vector<string> const& directories)
+    bool is_builtin(string const& file)
     {
+        string data;
+        string cmd = "type ";
+        cmd.append(file);
+        const int buffer_offset = 25;
+        const int max_buffer = buffer_offset + file.length();
+        char buffer[max_buffer];
+        FILE *stream = popen(cmd.c_str(), "r");
+        if (stream != nullptr) {
+            rewind(stream);
+            if (fgets(buffer, max_buffer, stream) != NULL){
+                data.append(buffer);
+            }
+            pclose(stream);
+        }
+
+        return boost::algorithm::contains(data, "builtin");
```

Comment:
remove `boost::algorithm::` as namespace `boost::algorithm::` is already imported at Line31